addon-resizer/nanny/nanny_lib.go

```diff
@@ -100,15 +118,24 @@ func PollAPIServer(k8s KubernetesClient, est ResourceEstimator, contName string,
 		expResources := est.scaleWithNodes(num)
 
 		// If there's a difference, go ahead and set the new values.
-		if !shouldOverwriteResources(int64(threshold), resources.Limits, resources.Requests, expResources.Limits, expResources.Requests) {
+		overwrite, op := shouldOverwriteResources(int64(threshold), resources.Limits, resources.Requests, expResources.Limits, expResources.Requests)
+		if !overwrite {
 			log.V(4).Infof("Resources are within the expected limits. Actual: %+v Expected: %+v", *resources, *expResources)
 			continue
 		}
 
+		now := time.Now()
+		if (op == scaleDown && now.Before(lastChange.Add(scaleDownDelay))) ||
+			(op == scaleUp && now.Before(lastChange.Add(scaleUpDelay))) {
+			log.V(3).Infof("Resources are not within the expected limits, Actual: %+v, accepted range: %+v. Skipping resource update because of scale up/down delay", *resources, *expResources)
+			continue
+		}
+
 		log.Infof("Resources are not within the expected limits, updating the deployment. Actual: %+v Expected: %+v", *resources, *expResources)
```

Comment:
nit: can you make this log consistent with the one above? Also could you add a log verbosity level?

Reply:
Log message has been updated. I have not added log verbosity because I believe that this message will be logged only when actual resize will be performed and therefore it is ok to log this message without a verbosity level being set.